library/spdm_responder_lib/libspdm_rsp_chunk_get.c

```diff
@@ -0,0 +1,151 @@
+#include "internal/libspdm_responder_lib.h"
+
+#if LIBSPDM_ENABLE_CHUNK_CAP
+
+libspdm_return_t libspdm_get_response_chunk_get(
+    void* context,
+    size_t request_size,
+    const void* request,
+    size_t* response_size,
+    void* response)
+{
+    libspdm_context_t* spdm_context;
+    libspdm_chunk_info_t* get_info;
+
+    const spdm_chunk_get_request_t* spdm_request;
+    spdm_chunk_response_response_t* spdm_response;
+
+    uint8_t* spdm_chunk;
+
+    spdm_context = (libspdm_context_t*) context;
+    spdm_request = (const spdm_chunk_get_request_t*) request;
+    spdm_response = (spdm_chunk_response_response_t*) response;
+    get_info = &spdm_context->chunk_context.get;
+
+    if ((spdm_context->local_context.capability.flags &
+         SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP) == 0) {
+        return libspdm_generate_error_response(
+            spdm_context,
+            SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+            response_size, response);
+    }
+
+    if (spdm_context->response_state != LIBSPDM_RESPONSE_STATE_NORMAL) {
+        return libspdm_responder_handle_response_state(
+            spdm_context,
+            spdm_request->header.request_response_code,
+            response_size, response);
+    }
+
+    if (spdm_context->connection_info.connection_state <
+        LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES) {
+        return libspdm_generate_error_response(
+            spdm_context,
+            SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+            response_size, response);
+    }
+
+    if (request_size < sizeof(spdm_chunk_get_request_t)) {
+        return libspdm_generate_error_response(
+            spdm_context,
+            SPDM_ERROR_CODE_INVALID_REQUEST, 0,
+            response_size, response);
+    }
+
+    /* TBD: Do we need to limit this command to 1.2? */
+    if (spdm_request->header.spdm_version
+        != libspdm_get_connection_version(spdm_context)) {
+        return libspdm_generate_error_response(
+            spdm_context,
+            SPDM_ERROR_CODE_VERSION_MISMATCH, 0,
+            response_size, response);
+    }
+
+    if (get_info->chunk_in_use == false) {
```

Comment:
The specification is vague on this, but I believe the intent is that following a `LargeResponse` error response the Requester can only issue `GET_VERSION` or `CHUNK_GET`. Assuming that is so the Responder needs to check that `get_info->chunk_in_use == false` for all the other Requests; `GET_DIGESTS`, etc. I'll file an issue to clarify the specification.